src/components/icon/icon.spec.ts

```diff
@@ -0,0 +1,476 @@
+        return builder.createAsync(MdIconLigatureTestApp).then((fixture) => {
+          const testComponent = fixture.debugElement.componentInstance;
+          const mdIconElement = fixture.debugElement.nativeElement.querySelector('md-icon');
+          testComponent.iconName = 'home';
+          fixture.detectChanges();
+          expect(mdIconElement.getAttribute('aria-label')).toBe('home');
+
+          testComponent.iconName = 'hand';
+          fixture.detectChanges();
+          expect(mdIconElement.getAttribute('aria-label')).toBe('hand');
+
+          done();
+        });
+      });
+
+      it('should use alt tag if aria label is not specified', (done: () => void) => {
+        return builder.createAsync(MdIconLigatureWithAriaBindingTestApp).then((fixture) => {
+          const testComponent = fixture.debugElement.componentInstance;
+          const mdIconElement = fixture.debugElement.nativeElement.querySelector('md-icon');
+          testComponent.iconName = 'home';
+          testComponent.altText = 'castle';
+          fixture.detectChanges();
+          expect(mdIconElement.getAttribute('aria-label')).toBe('castle');
+
+          testComponent.ariaLabel = 'house';
+          fixture.detectChanges();
+          expect(mdIconElement.getAttribute('aria-label')).toBe('house');
+
+          done();
+        });
+      });
+
+      it('should use provided aria label rather than icon name', (done: () => void) => {
+        return builder.createAsync(MdIconLigatureWithAriaBindingTestApp).then((fixture) => {
+          const testComponent = fixture.debugElement.componentInstance;
+          const mdIconElement = fixture.debugElement.nativeElement.querySelector('md-icon');
+          testComponent.iconName = 'home';
+          testComponent.ariaLabel = 'house';
+          fixture.detectChanges();
+          expect(mdIconElement.getAttribute('aria-label')).toBe('house');
+          done();
+        });
+      });
+
+      it('should use provided aria label rather than font icon', (done: () => void) => {
+        return builder.createAsync(MdIconCustomFontCssTestApp).then((fixture) => {
+          const testComponent = fixture.debugElement.componentInstance;
+          const mdIconElement = fixture.debugElement.nativeElement.querySelector('md-icon');
+          testComponent.fontSet = 'f1';
+          testComponent.fontIcon = 'house';
+          testComponent.ariaLabel = 'home';
+          fixture.detectChanges();
+          expect(mdIconElement.getAttribute('aria-label')).toBe('home');
+          done();
+        });
+      });
+    });
+  });
+}
+
+/** Test components that contain an MdIcon. */
+@Component({
+  selector: 'test-app',
+  template: `<md-icon>{{iconName}}</md-icon>`,
+  directives: [MdIcon],
+})
+class MdIconLigatureTestApp {
+  ariaLabel: string = null;
+  iconName = '';
+}
+
+@Component({
+  selector: 'test-app',
+  template: `<md-icon aria-label="{{ariaLabel}}" alt="{{altText}}">{{iconName}}</md-icon>`,
```

Comment:
One last comment:
for all these tests where you're using the `{{ }}` inside the attribute value, they should instead use the `[ ]` syntax, e.g.:
```
<md-icon [attr.aria-label]="ariaLabel" [alt]="altText">{{iconName}}</md-icon>
```